auth.php

```diff
@@ -265,6 +250,24 @@ public function enrol_user(string $enrolkey, bool $notify = true) : array {
                 $errors[$enrolplugin->courseid] = $enrol->can_self_enrol($enrolplugin);
             }
         }
+
+        if ($availableenrolids) {
```

Comment:
can this be `if(!empty($availableenrolids))` because its an array - makes it clearer rather than a loose comparison

Reply:
Changed this from feedback.